### ISSUE TITLE
Can the LoRA extract script be improved for lesser RAM usage?

### ISSUE THREAD

**Issue by FurkanGozukara** [kohya-ss/sd-scripts#1059]
Currently on Kaggle, with 29 GB ram and 15 GB VRAM it is failing to extract LoRA from SDXL DreamBooth models for 128x128 rank regular LoRA

I don't think so it should require this big requirements

**Comment by suede299** [comment]
Every time I extract the lora I have to select the cpu Device and then manually set the virtual memory to a very high level.

**Comment by FurkanGozukara** [comment]
> Every time I extract the lora I have to select the cpu Device and then manually set the virtual memory to a very high level.

yes currently using huge RAM. I told the Kohya and he said he could load 1 model into GPU and 1 model into RAM to reduce this. makes sense

**Comment by FurkanGozukara** [comment]
> --load_precision

thank you so much. if model is BF16 --load_precision we shouldnt use right?

**Comment by FurkanGozukara** [comment]
> If the model is bf16, `--load_precision bf16` can reduce memory usage without losing precision.

awesome ty
